_docs-sources/iac/usage/composing-your-own-service.md

```diff
@@ -0,0 +1,193 @@
+
+```hcl title=gw_service_guide/serverless-api/outputs.tf
+output "api_endpoint" {
+  description = "The URI of the API."
+  value       = module.api.api_endpoint
+}
+```
+
+## Using the service
+
+Now that you have defined the service, you can reference the service to create the resources in AWS.
+
+### Create the basic file structure
+
+First, create the files that will contain the reference to the service. Typically, you would create a module in one repository, then reference it in a different repository. For this tutorial, we’ll just create the reference in the top level directory for the sake of simplicity.
+
+Create a file called main.tf, which will contain a reference to the module, a directory called `/src`, which will contain all source code for the Lambda function, and a file called main.py, which will contain the Lambda function code.
+
+```bash
+touch gw_service_guide/main.tf
+
+mkdir -p gw_service_guide/src
+touch gw_service_guide/src/main.py
+```
+
+### Define the reference
+
+Define a module block in `gw_service_guide/main.tf` that references the relative location of the `serverless-api` service definition for the `source` attribute. In this guide, we are defining a Lambda function running Python 3.9, with the source path pointing to our `/src` directory, and the handler function `lambda_handler` defined in `gw_service_guide/src/main.py`.
+
+```hcl title=gw_service_guide/main.tf
+module "serverless_api" {
+  source = "./serverless-api/"
+
+  name               = "gw-guide-serverless-api"
+  lambda_runtime     = "python3.9"
+  lambda_source_path = "${path.module}/src"
+  lambda_handler     = "main.lambda_handler"
+}
+
+output "api_endpoint" {
+  value = module.serverless_api.api_endpoint
+}
+```
+
+Next, write a Python function that returns a status code of 200 and a response body stating "Hello from Gruntwork!". The response format is required for API Gateway to successfully return the response, to learn more refer to the [Lambda function response format documentation](https://docs.aws.amazon.com/apigateway/latest/developerguide/http-api-develop-integrations-lambda.html#http-api-develop-integrations-lambda.response).
+
+```python title=gw_service_guide/src/main.py
+import json
+
+def lambda_handler(event, context):
+    body = {"message": "Hello from Gruntwork!"}
+    return {"statusCode": 200, "body": json.dumps(body)}
+```
+
+Now that the service and references are defined, we can run `plan` and `apply` to create the resources in AWS.
```

Comment:
Is this sentence needed given we say basically the same thing on the very next line?

Reply:
Probably not. I was looking at the page at it looked empty, so I tried to compensate. I think it can go.

Reply:
Resolved in https://github.com/gruntwork-io/docs/pull/846/commits/822ebcde2f9c781a89b82f0786e8fa0da7bc9728